spec/models/subject_spec.rb

```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+RSpec.describe Subject, type: :model do
+  context 'when validating presence' do
+    it { is_expected.to validate_presence_of(:name) }
+    it { is_expected.to validate_presence_of(:course_id) }
```

Comment:
No need to validate the foreign key. You already have the test for the associations